### PR TITLE
functionalization: update wrapper to propagate symints

### PR DESCRIPTION
**Pull request by bdhirsh**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #75527
* #76358
* __->__ #78820
* #78819

Tested in the next PR by LTC tests (they were failing before). It's no longer enough just to copy the size/stride data from the inner tensor - if it has a custom sizes/strides policy, we need to make sure that we invoke the inner tensor's custom function, in order for functional tensors to properly "propagate" symbolic shapes

Differential Revision: [D37003884](https://our.internmc.facebook.com/intern/diff/D37003884)

### CODE REVIEW

File: aten/src/ATen/FunctionalTensorWrapper.h [hunk 1]
```diff
@@ -96,6 +96,16 @@ struct TORCH_API FunctionalTensorWrapper : public c10::TensorImpl {
 
   ~FunctionalTensorWrapper() override = default;
 
+  // FunctionalTensorWrapper overrides all custom size/stride function,
+  // so that if the inner tensor has a custo implementation
```
Comment (Krovatkin) [inline aten/src/ATen/FunctionalTensorWrapper.h:100]: `custo`m a typo

File: aten/src/ATen/FunctionalTensorWrapper.cpp [hunk 2]
```diff
@@ -260,6 +260,25 @@ const char* FunctionalTensorWrapper::tensorimpl_type_name() const {
     return "FunctionalTensorWrapper";
 }
 
+at::IntArrayRef FunctionalTensorWrapper::sizes_custom() const {
```
Comment (Krovatkin) [inline aten/src/ATen/FunctionalTensorWrapper.cpp:263]: is there a reason you don't want to call `value_.unsafeGetTensorImpl()->sizes_custom()`, that could save us a check?

Comment (bdhirsh) [reply aten/src/ATen/FunctionalTensorWrapper.cpp:263]: Aha I was gonna add a comment - we could do that and save ourselves the extra check, but we'd need to make all the `*_custom` functions public

**Review by ezyang — APPROVED** [review]

**Comment by bdhirsh** [comment]
@bdhirsh has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37003884).

**Comment by bdhirsh** [comment]
@bdhirsh has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37003884).

**Pull request closed without merging**